standard/statements.md

```diff
@@ -569,39 +567,14 @@ Multiple labels are permitted in a *switch_section*.
 > ```
 > *end note*
 
-> *Example*: The governing type of a `switch` statement can be the type `string`. For example:
-> ```csharp
-> void DoCommand(string command) {
->     switch (command.ToLower()) {
->         case "run":
->             DoRun();
->             break;
->         case "save":
->             DoSave();
->             break;
->         case "quit":
->             DoQuit();
->             break;
->         default:
->             InvalidCommand(command);
->             break;
->     }
-> }
-> ```
-> *end example*
-
-> *Note*: Like the string equality operators ([§12.11.8](expressions.md#12118-string-equality-operators)), the `switch` statement is case sensitive and will execute a given switch section only if the switch expression string exactly matches a `case` label constant. *end note*
-When the governing type of a `switch` statement is `string` or a nullable value type, the value `null` is permitted as a `case` label constant.
-
 The *statement_list*s of a *switch_block* may contain declaration statements ([§13.6](statements.md#136-declaration-statements)). The scope of a local variable or constant declared in a switch block is the switch block.
 
 Within a switch block, the meaning of a name used in an expression context shall always be the same ([§12.7.3.2](expressions.md#12732-invariant-meaning-in-blocks)).
 
 The statement list of a given switch section is reachable if the `switch` statement is reachable and at least one of the following is true:
 
-- The switch expression is a non-constant value.
-- The switch expression is a constant value that matches a `case` label in the switch section.
-- The switch expression is a constant value that doesn't match any `case` label, and the switch section contains the `default` label.
+- The switch expression value matches a `case` label in the switch section.
```

Comment:
This is wrong... no, it isn't even wrong.  It is trying to define a compile-time property (statement list of a given switch section is reachable) based on a runtime property (the switch expression value matches a `case` label in the switch section).

Reply:
More work needed.

Reply:
@gafter I think I see my mistake. Here's the whole bullet list with the bits **I deleted** struck-through:

> The statement list of a given switch section is reachable if the `switch` statement is reachable and at least one of the following is true:
> - <del>The switch expression is a non-constant value.</del>
> - The switch expression <del>is a constant</del> value <del>that</del> matches a `case` label in the switch section.
> - The switch expression <del>is a constant</del> value <del>that</del> doesn't match any `case` label, and the switch section contains the `default` label.
> - A switch label of the switch section is referenced by a reachable `goto case` or `goto default` statement.

If I restore the deletions of the 2nd and 3rd bullet items, we're only talking about a constant value matching a *constant_pattern*, and that can be done at compile-time, right?

I'm thinking I should also restore the 1st bullet item, no?

Reply:
I don't think that's right.  I think you've defined "matches" as something that is determined at run-time, not at compile-time.